### DISCUSSION TITLE
MapControl - How to use with the single project architecture

### DISCUSSION THREAD

**Discussion by trydalch** [unoplatform/uno#17890] [Q&A] [answered]
[The documentation for MapControl](https://platform.uno/docs/articles/controls/map-control-support.html) mentions adding the nuget package Uno.WinUI.Maps to each of the iOS and Android projects, then the XAML references Uno.Ui.Maps, and VS reports there's no MapResources found in the namespace. 

When I tried adding it to my project (single project architecture) the build fails saying that it can't build with both Uno.WinUI and Uno.UI packages referenced. I tried [following the guidance](https://platform.uno/docs/articles/uno-build-error-codes.html#unob0001-cannot-build-with-both-unowinui-and-unoui-nuget-packages-referenced) for resolving this UNOB0001 error, but have been unsuccessful thus far. 

Given some of the language in the MapControl documentation, I'm hoping it's just out of date (and that more features have been implemented).

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @trydalch,

thank you for asking this question. I have made some investigations and I couldn't get the package working with the current docs as well. I have opened an issue https://github.com/unoplatform/uno/issues/17904 to track this further. Please feel free to add more details in the comments. In the meantime I would recommend trying the https://github.com/Mapsui/Mapsui which supports Uno, it can be a nice alternative.
